Framework/HistogramData/src/HistogramItem.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MantidHistogramData/HistogramItem.h"
+#include "MantidHistogramData/Histogram.h"
+#include <algorithm>
+#include <utility>
+
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::Points;
+
+HistogramItem::HistogramItem(const Histogram &histogram, const size_t index)
+    : m_histogram(histogram), m_index(index) {}
+
+double HistogramItem::counts() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Counts) {
+    return histogramRef().counts()[m_index];
+  } else {
+    const Frequencies frequency{histogramRef().frequencies()[m_index]};
+    return Counts(frequency, binEdges())[0];
+  }
+}
+
+double HistogramItem::countVariance() const {
+  return histogramRef().countVariances()[m_index];
```

Comment:
This can be terribly expensive, computing all the variances. Same goes for several of the other methods below.